File: SDKs/zuora.com/2021-08-20/typescript/src/sdk/models/operations/getrampmetricsbysubscriptionkey.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetRampMetricsBySubscriptionKeyPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=subscriptionKey" })
  subscriptionKey: string;
}


export class GetRampMetricsBySubscriptionKeyHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Zuora-Entity-Ids" })
  zuoraEntityIds?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Zuora-Track-Id" })
  zuoraTrackId?: string;
}


export class GetRampMetricsBySubscriptionKeyRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetRampMetricsBySubscriptionKeyPathParams;

  @SpeakeasyMetadata()
  headers: GetRampMetricsBySubscriptionKeyHeaders;
}


export class GetRampMetricsBySubscriptionKeyResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  getRampMetricsBySubscriptionKeyResponseType?: shared.GetRampMetricsBySubscriptionKeyResponseType;

  @SpeakeasyMetadata()
  headers: Record<string, string[]>;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/zuora.com/2021-08-20/python/src/sdk/models/operations/get_rampmetricsbysubscriptionkey.py
import dataclasses
from typing import Optional
from ..shared import getrampmetricsbysubscriptionkeyresponsetype as shared_getrampmetricsbysubscriptionkeyresponsetype


@dataclasses.dataclass
class GetRampMetricsBySubscriptionKeyPathParams:
    subscription_key: str = dataclasses.field(metadata={'path_param': { 'field_name': 'subscriptionKey', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetRampMetricsBySubscriptionKeyHeaders:
    zuora_entity_ids: Optional[str] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'Zuora-Entity-Ids', 'style': 'simple', 'explode': False }})
    zuora_track_id: Optional[str] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'Zuora-Track-Id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetRampMetricsBySubscriptionKeyRequest:
    headers: GetRampMetricsBySubscriptionKeyHeaders = dataclasses.field()
    path_params: GetRampMetricsBySubscriptionKeyPathParams = dataclasses.field()
    

@dataclasses.dataclass
class GetRampMetricsBySubscriptionKeyResponse:
    content_type: str = dataclasses.field()
    headers: dict[str, list[str]] = dataclasses.field()
    status_code: int = dataclasses.field()
    get_ramp_metrics_by_subscription_key_response_type: Optional[shared_getrampmetricsbysubscriptionkeyresponsetype.GetRampMetricsBySubscriptionKeyResponseType] = dataclasses.field(default=None)
    
